synapse/handlers/user_directory.py

```diff
@@ -26,10 +26,12 @@
 from synapse.types import get_localpart_from_id
 from synapse.util.metrics import Measure
 
+from .state_deltas import StateDeltasHandler
+
 logger = logging.getLogger(__name__)
 
 
-class UserDirectoryHandler(object):
+class UserDirectoryHandler(StateDeltasHandler):
```

Comment:
composition > inheritance for this sort of thing, imho

Reply:
sure, that could be a good refactor later. Right now, I'm just splitting up what Matthew wrote.

Reply:
I don't think it will happen later. I'm not too bothered about this particular nit, but as a general principle, I don't think that we should be merging stuff into develop which wouldn't normally pass review just because Matthew wrote it in a hurry. 

I'm sorry if you've been given a pile of code to try and sort out, but I am strongly averse to adding to our technical debt by adding essentially unreviewed code to develop.

Reply:
Sure, that's why this code is being reviewed. And, yes, I am a staunch believer in the composition > inheritance thing, which -- if I wrote it -- I might well have investigated how to do it. But, I didn't write it, but the code works, and it matches the rest of the inheritance-based storage/handler APIs, so there's no reason going back and thinking about how to make this API composition-friendly if we're not doing it across everything else that's inheritance based.

Reply:
my impression was that changing to composition here would be approximately a three line change, but if it's more complicated than that, I'm happy to concede.